flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -1853,6 +1853,7 @@ private <OUT> DataStreamSource<OUT> createFileInput(
 
         SingleOutputStreamOperator<OUT> source =
                 addSource(monitoringFunction, sourceName, null, boundedness)
+                        .setMaxParallelism(CoreOptions.DEFAULT_PARALLELISM.defaultValue())
```

Comment:
I believe using the default value here is not correct. Imagine we are changing the default parallelism to 10 for some weird reason in the future, then this source would break.

Maybe introduce a new constant `ContinuousFileMonitoringFunction.MAX_PARALLELISM = 1` ?

I also think it would be good to add a comment explaining why this is necessary.

Reply:
-- Maybe introduce a new constant ContinuousFileMonitoringFunction.MAX_PARALLELISM = 1
Maybe [SingleOutputStreamOperator#forceNonParallel](https://github.com/apache/flink/blob/master/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/SingleOutputStreamOperator.java#L206) is more appropriate.

-- I also think it would be good to add a comment explaining why this is necessary.
Done.

Reply:
forceNonParallel is nice, good catch!